packages/x-charts/src/ChartsTooltip/useItemTooltip.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { InteractionContext, ItemInteractionData } from '../context/InteractionProvider';
+import { useSeries } from '../hooks/useSeries';
+import { useCartesianContext } from '../context/CartesianProvider';
+import { ZAxisContext } from '../context/ZAxisContextProvider';
+import { useColorProcessor } from '../context/PluginProvider/useColorProcessor';
+import {
+  ChartSeriesDefaultized,
+  ChartSeriesType,
+  ChartsSeriesConfig,
+} from '../models/seriesType/config';
+import { getLabel } from '../internals/getLabel';
+import { CommonSeriesType } from '../models/seriesType/common';
+
+export interface UseItemTooltipReturnValue<T extends ChartSeriesType> {
+  identifier: ItemInteractionData<T>;
+  color: string;
+  label: string | undefined;
+  value: ChartsSeriesConfig[T]['valueType'];
+  formattedValue: string | undefined;
+}
+
+export function useItemTooltip<T extends ChartSeriesType>(): null | UseItemTooltipReturnValue<T> {
```

Comment:
Should we use this hook ourselves?

Reply:
I tried it, but it would be a mess because the `slots.itemContent` is in the middle of this hook.

It expects to get the `series` and `getColor` whereas the hook directly provides the item `value` and `color`

Reply:
I don't get it, since we are not using any input props nor new contexts here, I would guess this could be used anywhere in the stack 🤔

Reply:
You have 3 nested components

- ChartsTooltip
- ChartsItemTooltipRender
- DefaultChartsItemTooltipContent

The first one gets the `item` identifier from the context
The second one gets the relevant axes and series from the context
The last one computes the color, formatted label, and formatted value and render the content

The hook is made to replace all of them. So it returns the computed values (the color, the formatted label, ...)

But we currently allow users to pass a slot to replace `DefaultChartsItemTooltipContent` which expect rought data (the defaultized series and a getColor callback)

So to do that we would need to extend the hook returned values as follow:


```diff
useItemTooltip: () => {
   color: string;
   formattedLabel: string;
   formattedValue: string;
+   /**
+   * @deprecated The needed information from the series item are already available from other attributes.
+   * /
+  series: DefaultizedSeries
+   /**
+   * @deprecated The color of the item is already available with `color` attribute
+   * /
+  getColor: (dataIndex: number) => string
}
```

Such that we could do something like

```jsx
function ChartsItemTooltip(){
  const data = useItemTooltip();

  const Content = slotProps.itemContent ?? DefaultChartsItemTooltip;
  return <Content {...data} />
}
```

Reply:
We could have an internal with that signature and a public one which filters out the unnecessary values. Then when we simplify customisation we can move to using the public one.

Reply:
What would be the interest? It sounds like an extra effort for something we are nearly sure to remove in the next major

Reply:
The main idea is that, if we use public apis internally, then it is easier to maintain and track implementation issues. But it is reasonable to do that in a future version 😅